MobiusCore/Test/EffectHandlers/EffectRouterDSLTests.swift

```diff
@@ -36,29 +36,92 @@ private enum Event: Equatable {
 class EffectRouterDSLTests: QuickSpec {
     // swiftlint:disable function_body_length
     override func spec() {
+        context("An EffectHandler which always ends as soon as it is called") {
+            var wasDisposed: Bool!
+            var connection: Connection<Int>!
+
+            beforeEach {
+                wasDisposed = false
+                connection = EffectRouter<Int, ()>()
+                    .routeEffects(equalTo: 1).to { _, response in
+                        response.end()
+                        return AnonymousDisposable {
+                            wasDisposed = true
+                        }
+                    }
+                    .asConnectable
+                    .connect { _ in }
+            }
+
+            it("should not be disposed if `end()` was called") {
+                connection.accept(1)
+                connection.dispose()
+
+                expect(wasDisposed).to(beFalse())
+            }
+
+            it("should be disposed if the connection is disposed before `end()` is called") {
+                connection.dispose()
+
+                expect(wasDisposed).to(beFalse())
+            }
+        }
+
+        context("An EffectHandler which calls end after some time") {
+            var wasDisposed: Bool!
+            var connection: Connection<Int>!
+            var end: (() -> Void)!
+
+            beforeEach {
+                wasDisposed = false
+                end = { fail("End should have been set") }
+                connection = EffectRouter<Int, ()>()
+                    .routeEffects(equalTo: 1).to { _, response in
```

Comment:
I guess `response` should be changed to `callback` everywhere in this file (or pretty much).